app/code/Magento/Sales/Controller/Adminhtml/Invoice/AbstractInvoice/PrintAction.php

```diff
@@ -55,13 +55,10 @@ public function execute()
             )->get($invoiceId);
             if ($invoice) {
                 $pdf = $this->_objectManager->create(\Magento\Sales\Model\Order\Pdf\Invoice::class)->getPdf([$invoice]);
-                $date = $this->_objectManager->get(
-                    \Magento\Framework\Stdlib\DateTime\DateTime::class
-                )->date('Y-m-d_H-i-s');
                 return $this->_fileFactory->create(
-                    'invoice' . $date . '.pdf',
+                    "invoice_$invoiceId.pdf",
```

Comment:
Would it better to filter `$invoiceId` before?
I am concerned about a potential path traversal attack.
I suppose changing line 51 would be just fine: `$invoiceId = (int) $this->getRequest()->getParam('invoice_id');`.

Reply:
Thanks, Riccardo. That's a very good point.